test/shared/src/main/scala/zio/test/MutableRunnableSpec.scala

```diff
@@ -0,0 +1,144 @@
+package zio.test
+
+import zio.clock.Clock
+import zio.duration._
+import zio.test.environment.TestEnvironment
+import zio.{Chunk, Has, URIO, ZIO, ZLayer}
+
+import scala.util.control.NoStackTrace
+
+/**
+ * Syntax for writing test like
+ * {{{
+ * object MySpec extends MutableRunnableSpec(layer) {
+ *   suite("foo") {
+ *     testM("name") {
+ *     } @@ ignore
+ *
+ *     test("name 2")
+ *   }
+ *   suite("another suite") {
+ *     test("name 3")
+ *   }
+ * }
+ * }}}
+ */
+class MutableRunnableSpec[R <: Has[_]](layer: ZLayer[TestEnvironment, Throwable, R])
```

Comment:
There is some overlap here with what Balasz is doing.

Reply:
Yes, I will speak with him. Maybe renaming would be good to call this 
DefaultMutableRunnableSpec -> MutableRunnableSpec
DefaultMutableRunnableSpec -> CustomMutableRunnableSpec
to have it consistent. What do you think? @jdegoes

Reply:
Maybe so.